modules/agent-kafka/src/main/scala-2.13/io/janstenpickle/trace4cats/agent/kafka/AgentKafka.scala

```diff
@@ -46,21 +46,26 @@ object AgentKafka
     bufferSize: Int,
     trace: Boolean,
     traceRate: Option[Double]
-  ): IO[ExitCode] = (for {
-    implicit0(logger: Logger[IO]) <- Resource.eval(Slf4jLogger.create[IO])
-
-    kafkaExporter <- AvroKafkaSpanExporter[IO, Chunk](kafkaBootstrapServers, kafkaTopic)
-  } yield CommonAgent.run[IO](
-    port,
-    bufferSize,
-    "Avro Kafka",
-    Map[String, AttributeValue](
-      "bootstrap.servers" -> AttributeValue.StringList(kafkaBootstrapServers),
-      "topic" -> kafkaTopic
-    ),
-    kafkaExporter,
-    s"Kafka topic '$kafkaTopic'",
-    trace,
-    traceRate
-  )).use(identity)
+  ): IO[ExitCode] =
+    Resource
+      .eval(Slf4jLogger.create[IO])
+      .flatMap { implicit logger: Logger[IO] =>
```

Comment:
This pattern is so repetitive, I even would like to have syntax extensions for the `Slf4jLogger` object:
```scala
Slf4jLogger.provide[F](implicit logger => something)  // something: F[A]
Slf4jLogger.provideResource[F](implicit logger => something)  // something: Resource[F, A]
```

I have something similar in my private codebase.

@janstenpickle wdyt?

Reply:
But maybe it's not worth it...

Reply:
Maybe something that could be contributed to log4cats directly?

Reply:
nevermind. now it's a one-off change. further changes are not as often.